docs/src/routes/docs/components/snackbar.svx

```diff
@@ -0,0 +1,89 @@
+---
+name: Snackbar
+relatedComponents:
+  - name: SnackbarContainer
+    link: ./docs/components/snackbar-container
+shadowRaised: '0 2px 4px -1px rgba(0, 0, 0, .2), 0 3px 5px 0 rgba(0, 0, 0, .14), 0 1px 10px 0 rgba(0, 0, 0, .12);'
+---
+
+<script>
+  import { Snackbar } from 'attractions';
+  import Showcase from 'src/containers/docs/showcase.svelte';
+  import ColorPreview from 'src/components/docs/color-preview.svelte';
+  import ShadowPreview from 'src/components/docs/shadow-preview.svelte';
+  import RelatedComponents from 'src/components/docs/related-components.svelte';
+</script>
+
+# Snackbar
+
+A small snackbar to display status messages.
+
+<RelatedComponents components={relatedComponents} class="mb" />
+
+Note: below is not the intended way of using this component. It is meant to be passed into the `showSnackbar` function of the [==`SnackbarContainer`==](./docs/components/snackbar-container) component.
+
+<Showcase>
+  <div slot="showcase" class="padded">
+    <Snackbar
+      text="Hello!"
+      action={{
+        text: 'oh, hi',
+        callback: () => console.log('ack'),
+      }}
+      closeCallback={() => {}}
+    />
+  </div>
+  <div slot="source">
+
+```svelte
+<Snackbar
+  text="Hello!"
+  action={{
+    text: 'oh, hi',
+    callback: () => console.log('ack'),
+  }}
+  closeCallback={() => {}}
+/>
+```
+
+  </div>
+</Showcase>
+
+
+## Properties {#properties}
+
+### Functional Properties
+
+| Name | Default | Type | Description |
+| ---- | ------- | ---- | ----------- |
+| ==**`text`**==&nbsp;<span class="required">*</span> |  | `string` | The text to show on the snackbar. |
+| ==**`closeCallback`**==&nbsp;<span class="required">*</span> |  | `Function` | The callback to call to close the snackbar in the `SnackbarContainer`. Passed internally by the `SnackbarContainer` when using the `showSnackbar` function. |
+| **`action`** | `null` | `{ text, callback() }` | The action for the button on the snackbar. If this is `null`, the button is not rendered. Otherwise it has `text` as a label and calls `callback` on click. |
```

Comment:
We usually don't mark the types of object properties, why did you feel the need to write the `()` in `callback()` here?

Reply:
This is a very ugly table, honestly. Originally I wanted to have this cell say `{ text: string, callback: Function }`, but this is a huge blob of unwrappable text, and unfortunately, the default value column also had a huge blob of unwrappable text, and the whole table just went to shit, so I figured I'd at least indicate that the callback is a function without explicitly writing out the types

Reply:
I am all for specifying the types everywhere, but perhaps we need a better format for the table. For now, we can wrap the object (although that might be too many lines for each row) to look something like:
```typescript
{
    text: string,
    callback: () => void
}
```
It would be great if we use the Typescript notation, and even greater if we add type definitions file

Reply:
We'll hold on to full TypeScript notation until after we're done, but for now I did the wrapping